oak_functions/client/rust/src/lib.rs

```diff
@@ -48,14 +63,35 @@ impl Client {
         .context("Could not create Oak Functions client")?;
         Ok(Client { inner })
     }
-    pub async fn invoke(&mut self, request: Request) -> anyhow::Result<Response> {
-        let response = self
-            .inner
-            .send(request)
-            .await
-            .context("Error invoking Oak Functions instance")?;
-        response
-            .ok_or_else(|| anyhow::anyhow!("Empty response"))
-            .and_then(|rsp| Response::decode(rsp.as_ref()).context("Could not decode the response"))
+}
+
+#[async_trait]
+impl Session for Client {
+    async fn invoke(&mut self, request: Request) -> anyhow::Result<Response> {
+        handle_response(self.inner.send(request).await)
+    }
+}
+
```

Comment:
This doesn't define the `new` method (despite the shared type signature), as the compiler warns that not all futures involved implement the `Send` trait (which the async-trait macro requires by default. 

async-trait does offer a [non-thread safe option](https://crates.io/crates/async-trait), but not sure if it's safe to use. Probably fine to keep it like though.

Reply:
Yeah the current version seems fine, also I believe there is never a case in which the new method needs to be invoked from a dyn trait object, it's always called on the concrete type, so lifting it into this trait would probably not be appropriate anyways.

Reply:
Would potentially make the main.rs slightly nicer, but agree not strictly needed